driver-reactive-streams/src/main/com/mongodb/reactivestreams/client/internal/MongoCollectionImpl.java

```diff
@@ -694,7 +694,9 @@ public ListSearchIndexesPublisher<Document> listSearchIndexes() {
     @Override
     public <TResult> ListSearchIndexesPublisher<TResult> listSearchIndexes(final Class<TResult> resultClass) {
         notNull("resultClass", resultClass);
-        return new ListSearchIndexesPublisherImpl<>(mongoOperationPublisher.withDocumentClass(resultClass));
+
+        return new ListSearchIndexesPublisherImpl<>(mongoOperationPublisher
```

Comment:
I'm not clear on why, in driver-sync's MongoCollectionImpl, read concern seems to have been removed, while here it seems like we're adding it.  Can you clarify the intent?